src/nvim/popupmnu.c

```diff
@@ -852,7 +855,8 @@ void pum_recompose(void)
 /// Only valid when pum_visible() returns TRUE!
 int pum_get_height(void)
 {
-  return pum_height;
+  int ui_pum_height = ui_pum_get_height();
+  return ui_pum_height ? ui_pum_height : pum_height;
```

Comment:
use `pum_external && ui_pum_height` as the condition.

Reply:
Please tell me the reason to change the condition so. The reasons I think are:

1. When a `UI` that does not support `ext_popupmenu` uses `nvim_pum_set_height()`, it does not use its `pum_height`.
1.  To avoid unnecessary calls to `ui_pum_get_height()`.
1. To clarify the meaning of the code when others see this code in the future.

If it is 1, check that `ui->ui_ext[kUIpopupmenu]` is true in `nvim_set_pum_height()`.

In case of 2, change as follows:

```Diff
--- a/src/nvim/popupmnu.c
+++ b/src/nvim/popupmnu.c
@@ -855,8 +855,8 @@ void pum_recompose(void)
 /// Only valid when pum_visible() returns TRUE!
 int pum_get_height(void)
 {
-  int ui_pum_height = ui_pum_get_height();
-  return pum_external && ui_pum_height ? ui_pum_height : pum_height;
+  int ui_pum_height = pum_extara ? ui_pum_get_height() : 0;
+  return ui_pum_height ? ui_pum_height : pum_height;
 }

 void pum_set_boundings(dict_T *dict)
```

In the case of 3, write an appropriate comment. I think the above is more appropriate than changing the conditions.

Reply:
Well, the point is just what the code says: whenever `pum_external` is false external popupmenu is disabled, so `pum_height` should be used (which I guess implies all of the above). Though I agree just substituting it into the ternary doesn't look very good.  Could rewrite it like so:

```
int pum_get_height(void)
{
  if (pum_external) {
    int ui_pum_height = ui_pum_get_height();
    if (ui_pum_height) {
      return ui_pum_height;
    }
  }
  return pum_height;
}
```